lib/ansible/module_utils/ovirt.py

```diff
@@ -0,0 +1,544 @@
+#!/usr/bin/python
+# -*- coding: utf-8 -*-
+#
+# Copyright (c) 2016 Red Hat, Inc.
+#
+# This file is part of Ansible
+#
+# Ansible is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# Ansible is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with Ansible.  If not, see <http://www.gnu.org/licenses/>.
+#
+
+import inspect
+import time
+
+from abc import ABCMeta, abstractmethod
+from datetime import datetime
+from distutils.version import LooseVersion
+from enum import Enum
+
+try:
+    import ovirtsdk4 as sdk
+    import ovirtsdk4.types as otypes
+    import ovirtsdk4.version as sdk_version
+    HAS_SDK = LooseVersion(sdk_version.VERSION) >= LooseVersion('4.0.0')
+except ImportError:
+    HAS_SDK = False
+
+
+BYTES_MAP = {
+    'kib': 2**10,
+    'mib': 2**20,
+    'gib': 2**30,
+    'tib': 2**40,
+    'pib': 2**50,
+}
+
+
+def check_sdk(module):
+    if not HAS_SDK:
+        module.fail_json(
+            msg='ovirtsdk4 version 4.0.0 or higher is required for this module'
+        )
+
+
+def get_dict_of_struct(struct):
+    """
+    Convert SDK Struct type into dictionary.
+    """
+    def remove_underscore(val):
+        if val.startswith('_'):
+            val = val[1:]
+            remove_underscore(val)
+        return val
+
+    res = {}
+    if struct is not None:
+        for key, value in struct.__dict__.iteritems():
+            key = remove_underscore(key)
+            if value is None:
+                continue
+            elif isinstance(value, sdk.Struct):
+                res[key] = get_dict_of_struct(value)
+            elif isinstance(value, Enum) or isinstance(value, datetime):
+                res[key] = str(value)
+            elif isinstance(value, list):
+                res[key] = []
+                for i in value:
+                    if isinstance(i, sdk.Struct):
+                        res[key].append(get_dict_of_struct(i))
+                    elif isinstance(i, Enum):
+                        res[key].append(str(i))
+                    else:
+                        res[key].append(i)
+            else:
+                res[key] = value
+
+    return res
+
+
+def create_connection(auth):
+    """
+    Create a connection to Python SDK, from task `auth` parameter.
```

Comment:
In the pydoc for this function can you please add info about what keys in the dictionary are required?